tensorflow/python/ops/check_ops.py

```diff
@@ -2120,24 +2120,28 @@ def ensure_shape(x, shape, name=None):
   """Updates the shape of a tensor and checks at runtime that the shape holds.
 
   For example:
-  ```python
-  x = tf.compat.v1.placeholder(tf.int32)
-  print(x.shape)
-  ==> TensorShape(None)
-  y = x * 2
-  print(y.shape)
-  ==> TensorShape(None)
-
-  y = tf.ensure_shape(y, (None, 3, 3))
-  print(y.shape)
-  ==> TensorShape([Dimension(None), Dimension(3), Dimension(3)])
-
-  with tf.compat.v1.Session() as sess:
-    # Raises tf.errors.InvalidArgumentError, because the shape (3,) is not
-    # compatible with the shape (None, 3, 3)
-    sess.run(y, feed_dict={x: [1, 2, 3]})
 
-  ```
+  >>> # tf.placeholder() is not compatible with eager execution
```

Comment:
Why don't you write an example without placeholders, using tf.function and signatures instead?

Reply:
Yes that makes more sense. Will add that. 👍